src/metabase/actions/execution.clj

```diff
@@ -84,35 +93,29 @@
-                              (map (juxt (comp u/slugify :name) :name))
-                              (into {})
-                              (m/filter-keys (set (map :id parameters))))
-        _ (api/check (action/unique-field-slugs? table-fields)
-                     400
-                     (tru "Cannot execute implicit action on a table with ambiguous column names."))
-        _ (api/check (= (count pk-fields) 1)
-                     400
-                     (tru "Must execute implicit action on a table with a single primary key."))
-        extra-parameters (set/difference (set (keys request-parameters))
-                                         (set (keys slug->field-name)))
-        pk-field (first pk-fields)
-        simple-parameters (update-keys request-parameters (comp keyword slug->field-name))
-        pk-field-name (keyword (:name pk-field))
-        row-parameters (cond-> simple-parameters
-                         (not= implicit-action :row/create) (dissoc pk-field-name))
-        requires_pk (contains? #{:row/delete :row/update} implicit-action)]
+        table-fields                                (:fields table)
+        pk-fields                                   (filterv #(isa? (:semantic_type %) :type/PK) table-fields)
+        slug->field-name                            (->> table-fields
+                                                         (map (juxt (comp u/slugify :name) :name))
+                                                         (into {})
+                                                         (m/filter-keys (set (map :id parameters))))
+        _                                           (api/check (action/unique-field-slugs? table-fields)
+                                                      400
+                                                      (tru "Cannot execute implicit action on a table with ambiguous column names."))
+        _                                           (api/check (= (count pk-fields) 1)
+                                                      400
+                                                      (tru "Must execute implicit action on a table with a single primary key."))
+        extra-parameters                            (set/difference (set (keys request-parameters))
+                                                                    (set (keys slug->field-name)))
+        pk-field                                    (first pk-fields)
+        simple-parameters                           (update-keys request-parameters (comp keyword slug->field-name))
+        pk-field-name                               (keyword (:name pk-field))
+        row-parameters                              (cond-> simple-parameters
+                                                      (not= implicit-action :row/create) (dissoc pk-field-name))
+        requires_pk                                 (contains? #{:row/delete :row/update} implicit-action)]
```

Comment:
I'm no expert on whitespace, but this seems excessive? Too much whitespace makes things harder to read, not easier. This is a borderline case though, so I won't care if you ignore it